controllers/argocd/status.go

```diff
@@ -207,7 +207,10 @@ func (r *ReconcileArgoCD) reconcileStatusSSO(cr *argoproj.ArgoCD) error {
 func (r *ReconcileArgoCD) reconcileStatusPhase(cr *argoproj.ArgoCD) error {
 	var phase string
 
-	if cr.Status.ApplicationController == "Running" && cr.Status.Redis == "Running" && cr.Status.Repo == "Running" && cr.Status.Server == "Running" {
+	if ((!cr.Spec.Controller.IsEnabled() && cr.Status.ApplicationController == "Unknown") || cr.Status.ApplicationController == "Running") &&
```

Comment:
Instead of having a status of `Unknown` can we have a state called `Disabled`. It would be more intutive for the user to understand that the component is explicitly disabled.

Reply:
We do not maintain a `Disabled` state today and add `Unknown` as the state for missing components like `ApplicationSet` if `spec.applicationset = nil`. It would be like adding a new behavior and a new Status as part of bugfix.

Reply:
If not as part of this PR, we can create a separate PR for adding the `Disabled` state. That being a string and not part of the API spec, it can be introduced with minimal impact to end users.